src/components/shared/Countdown.tsx

```diff
@@ -0,0 +1,63 @@
+import { StyleSheet, View, Text, StyleProp, TextStyle } from "react-native";
+import * as React from "react";
+
+interface CountdownProps {
+  time: number; // in seconds
+  counter: number; // in seconds
+  style?: StyleProp<TextStyle>;
+  onFinish: () => void;
+}
+
+interface CountdownState {
+  timeRemaining: number;
+}
+
+export default class Countdown extends React.PureComponent<
+  CountdownProps,
+  CountdownState
+> {
+  static defaultProps = {
+    time: 30,
+    counter: 1000
+  };
+
+  state = {
+    timeRemaining: this.props.time
+  };
+
+  timer: any; // should be a number but TS is giving an error
+
+  componentDidMount() {
+    this.startCountdown();
+  }
+
+  startCountdown() {
+    const { counter, onFinish } = this.props;
+
+    this.timer = setInterval(() => {
+      this.setState({ timeRemaining: this.state.timeRemaining - 1 });
+      if (this.state.timeRemaining <= 0) {
```

Comment:
Just (this.state.timeRemaining) should do, right?

Reply:
This is a safer way. setInterval doesn't always gets executed at the exact time. (read event loop for that).

So there might be a case where the STI gets executed a bit later than expected and hence the difference might come negative.

Gist:

The time that you pass in setInterval or setTimeout is the MINIMUM time in which it gets executed not the exact time.

Case:
```js
setTimeout(() => console.log("a"), 1000)
setTimeout(() => console.log("b"), 1000)
setTimeout(() => console.log("c"), 1000)
```

If all of them got executed at the correct time, that means JS is multithreaded which actually it is not.

Reply:
`Thanks, I remember you had told about the callback, which browser(environment) gives back to JS thread.` (Event Loop Tick)